documentation/concepts.md

```diff
@@ -3,17 +3,114 @@ layout: page
 title: "Concepts"
 --- 
 
-firewalld has a two layer design: The core layer and the D-Bus layer on top. The core layer is responsible for handling the configuration and the back ends like iptables, ip6tables, ebtables, ipset and the module loader.
+## Introduction
 
-![firewalld-structure+nftables](firewalld-structure+nftables.png "firewalld structure")
-*firewalld structure*
+firewalld aims to be a simple, zoning, stateful firewall. Firewall rules
+are organized into zones and policies. Zones logically separate the
+network. Policies apply rules to traffic flowing between zones.
 
-The firewalld D-Bus interface is the primary way to alter and create the firewall configuration. The interface is used by all firewalld provided online tools, like for example firewall-cmd, firewall-config and firewall-applet. firewall-offline-cmd is not talking to firewalld, but altering and creating firewalld configuration files directly using the firewalld core with the IO handlers. firewall-offline-cmd can be used while firewalld is running, but it is not recommended as it is only able to alter the permanent configuration that are visible in firewall after about five seconds.
+This looks like the following:
 
-More information on the firewalld D-Bus API is available [here](man-pages/firewalld.dbus.html).
+```
++----------+     policyA     +----------+
```

Comment:
It appears that the concept is clearly outlined and is easily understood. How do you feel about adding styled graphs with the jekyll-mermaid plugin?
```mermaid
graph LR;
    A("internal<br>(zone)") -->|policyB| B("external<br>(zone)")
    B -->|policyA| A 
```

Reply:
Is the plugin enable by default?

Reply:
I found this blog post: https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/

But it's unclear to me if it's enabled in github pages, or just repo markdown files.

Reply:
I'll use the mermaid, if it doesn't render we'll figure it out after the merge. ;)

Reply:
> So it definitely took some playing around to get it to work. I ended up having to add some script tags to the `page` layout. I will probably tweak some settings.

It looks great, thanks!